src/NUnitFramework/framework/Internal/Execution/WorkItem.cs

```diff
@@ -248,6 +253,16 @@ public virtual void Execute()
 #endif
         }
 
+        private ManualResetEvent _completionEvent = new ManualResetEvent(false);
```

Comment:
`ManualResetEvent` is `IDisposable` and therefore the owning class should implement `IDisposable` and dispose it to free the kernel handles.

Reply:
The field should be `readonly` too.

Reply:
Good one!